docs/spec.md

```diff
@@ -132,6 +139,38 @@ Informs the client that channels are no longer available.
 }
 ```
 
+### Parameter Values
+
+Informs the client about parameters. Only supported if the server declares the `parameters` [capability](#server-info).
+
+#### Fields
+
+- `op`: string `"parameterValues"`
+- `parameters`: array of:
+  - `name`: string, name of the parameter
+  - `value`: number | boolean | string | number[] | boolean[] | string[]
+- `type`: string, one of
+  - `"serverInitiated"` - The message was sent by the server without being previously requested
+  - `"getParametersResponse"` - The message was sent as a response to the client's [Get Parameters request](#get-parameters)
+  - `"valueUpdate"` - The message was sent because one or more [subscribed](#subscribe-parameter-update) parameters have changed their value
```

Comment:
I don't understand this field, what is a client supposed to do with it? The id field below already handles linking requests/responses together for getParameters.

Reply:
I think this was answered above:

> For instance, if the client requests 2 parameters and the server replies with the corresponding 2 parameter values, the client should interpret this as the response to its previous request. It should not mean that there are only 2 parameters in total.

Reply:
> the client should interpret this as the response to its previous request

Isn't that what the `id` field is for?

Reply:
The question is, do we have to distinguish between server initiated messages and value updates (parameter subscription callbacks)? If so, then we need this extra field.

Reply:
We currently allow the server to send parameters at random points of time (https://github.com/foxglove/ws-protocol/pull/288#discussion_r1038506796). The server could e.g. send all available parameters to newly connected clients. If that's really done is up to the server implementation.

Reply:
What would the client code do?

```typescript
if (serverInitiated) {
 ... // ?
}
```

Reply:
This makes sense, although I wouldn't call it "server initiated" since it has a more specific meaning of "this is the complete parameter list, delete all parameters not included in this message". I don't think supporting server-initiated full parameter lists is an important goal; I recommend removing the `type` field completely.

Reply:
OK, I will do that